libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -193,22 +199,14 @@ proc getGossipPeers(g: GossipSub): Table[string, ControlMessage] {.gcsafe.} =
       if topic notin g.gossipsub:
         trace "topic not in gossip array, skipping", topicID = topic
         continue
-
-      while result.len < GossipSubD:
-        if g.gossipsub.getOrDefault(topic).len == 0:
-          trace "no peers for topic, skipping", topicID = topic
-          break
-
-        let id = toSeq(g.gossipsub.getOrDefault(topic)).sample()
-        if id in g.gossipsub.getOrDefault(topic):
-          g.gossipsub[topic].excl(id)
-          if id notin gossipPeers:
-            if id notin result:
-              result[id] = ControlMessage()
-            result[id].ihave.add(ihave)
-
-    libp2p_gossipsub_peers_per_topic_gossipsub
-      .set(g.gossipsub.getOrDefault(topic).len.int64, labelValues = [topic])
+      
+      var extraPeers = toSeq(g.gossipsub[topic])
```

Comment:
I'm not sure if this is making it better at all, for one this will be copied at least once every second, the previous algorithm didn't do that and it was doing the right thing already. Either way I wouldn't change this things unless there is a really good reason, because we can/will introduce additional subtle issues. Let's keep it as is right now.

Reply:
it was exactly the same before tho.... toSeq(x).sample